### PR TITLE
Take new (raytracing) termination instructions into account.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

**Comment by ehsannas** [comment]
Working on adding a unit test.

File: source/opt/reflect.h [hunk 1]
```diff
@@ -61,7 +61,9 @@ inline bool IsSpecConstantInst(SpvOp opcode) {
 }
 inline bool IsTerminatorInst(SpvOp opcode) {
   return (opcode >= SpvOpBranch && opcode <= SpvOpUnreachable) ||
-         (opcode == SpvOpTerminateInvocation);
+         (opcode == SpvOpTerminateInvocation ||
+          opcode == SpvOpTerminateRayKHR ||
+          opcode == SpvOpIgnoreIntersectionKHR);
```
Comment (s-perron) [inline source/opt/reflect.h:66]: Could this call `spvOpcodeIsBlockTerminator` instead of checking the opcode itself?

**Comment by greg-lunarg** [comment]
Without this code, what kind of error is incurred by spirv-opt: valid but unoptimimal code? valid but incorrect code? invalid code? spirv-opt abort?

**Comment by ehsannas** [comment]
I was just going to ping you folks via email. This could cause an error `OpLabel inside basic block` (as found [here](https://github.com/microsoft/DirectXShaderCompiler/issues/3285#issuecomment-738129117))or it could also result in a crash depending on the input shader.

spirv-opt had not taken these new termination instructions into account.

File: source/opt/reflect.h (hunk 1 above)
Comment (ehsannas) [reply source/opt/reflect.h]: yes, they are checking the same 10 opcodes. Done.

**Comment by greg-lunarg** [comment]
In other words, it will create invalid spirv, which could cause subsequent passes to crash.

**Review by dgkoch — APPROVED** [review]
Looks reasonable

**Review by s-perron — APPROVED** [review]

**Pull request merged**